packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_update_loan_scheme.test.ts

```diff
@@ -0,0 +1,345 @@
+import { DeFiDRpcError, GenesisKeys } from '@defichain/testcontainers'
+import { getProviders, MockProviders } from '../provider.mock'
+import { P2WPKHTransactionBuilder } from '../../src'
+import { fundEllipticPair, sendTransaction } from '../test.utils'
+import { WIF } from '@defichain/jellyfish-crypto'
+import BigNumber from 'bignumber.js'
+import { LoanMasterNodeRegTestContainer } from './loan_container'
+import { Testing } from '@defichain/jellyfish-testing'
+import { RegTest } from '@defichain/jellyfish-network'
+
+describe('loan.updateLoanScheme()', () => {
+  const container = new LoanMasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  let providers: MockProviders
+  let builder: P2WPKHTransactionBuilder
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+
+    providers = await getProviders(testing.container)
+    providers.setEllipticPair(WIF.asEllipticPair(GenesisKeys[GenesisKeys.length - 1].owner.privKey))
+    builder = new P2WPKHTransactionBuilder(providers.fee, providers.prevout, providers.elliptic, RegTest)
+
+    // Default scheme
+    await testing.container.call('createloanscheme', [100, new BigNumber(1.5), 'default'])
+    await testing.generate(1)
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  beforeEach(async () => {
+    // Fund 10 DFI UTXO
+    await fundEllipticPair(testing.container, providers.ellipticPair, 10)
+    await providers.setupMocks() // Required to move utxos
+
+    await testing.container.call('createloanscheme', [200, new BigNumber(2.5), 'scheme1'])
+    await testing.generate(1)
+  })
+
+  afterEach(async () => {
+    const data = await testing.container.call('listloanschemes')
+    const result = data.filter((d: { default: boolean }) => !d.default)
+
+    for (let i = 0; i < result.length; i += 1) {
+      // Delete all schemes except default scheme
+      await testing.container.call('destroyloanscheme', [result[i].id])
+      await testing.generate(1)
+    }
+  })
+
+  it('should updateLoanScheme', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.updateLoanScheme({
+      ratio: 300,
+      rate: new BigNumber(3.5),
+      identifier: 'scheme1',
+      update: new BigNumber('0xffffffffffffffff')
```

Comment:
what does `update: new BigNumber('0xffffffffffffffff')` do here ?

Reply:
It is the default value to indicate update -> updateLoanScheme
If I don't pass in any value, then this is to indicate create -> createLoanScheme

Reply:
this is `std::numeric_limits<uint64_t>::max()`. if passed with `LoanScheme` message, it means update at next possible block.

Reply:
a small comment/reference should be enough to explain it